### DISCUSSION TITLE
umi工程中如何在onRouteChange中获取initialState

### DISCUSSION THREAD

**Discussion by tomseanmy** [umijs/umi#11324] [Q&A] [answered]
使用场景是在onRouteChange函数中判断initialState是否包含登录信息，若不包含就指向登录页，现在无法在非antd-pro工程中的onRouteChange函数获取到initialState，请问有相应的解决方案吗？

**Comment by fz6m** [comment, marked as answer]
在 layout 里注册个 `history.listen` 一下，然后就可以用 model 数据了。

**Reply by tomseanmy** [reply]
非常感谢，用这个方法已经搞定了😁
